webpack/move_to_foreman/components/common/table/components/Table.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table as PfTable } from 'patternfly-react';
+import EmptyState from '../../emptyState';
+import PaginationRow from '../../../../../components/PaginationRow/index';
+
+import TableBody from './TableBody';
+
+const Table = ({
+  columns,
+  rows,
+  emptyState,
+  bodyMessage,
+  children,
+  itemCount,
+  pagination,
+  onPaginationChange,
+  ...props
+}) => {
+  if (rows.length === 0 && bodyMessage === undefined) {
+    return <EmptyState {...emptyState} />;
+  }
+
+  const shouldRenderPagination = itemCount && pagination;
+
+  const body = children || [
+    <PfTable.Header key="header" />,
+    <TableBody key="body" columns={columns} rows={rows} message={bodyMessage} rowKey="id" />,
+  ];
+
+  return (
+    <div>
+      <PfTable.PfProvider
+        columns={columns}
+        className="table-fixed"
+        striped
+        bordered
+        hover
+        {...props}
+      >
+        {body}
+      </PfTable.PfProvider>
+      {shouldRenderPagination && (
+        <PaginationRow
+          viewType="table"
+          itemCount={itemCount}
+          pagination={pagination}
+          onChange={onPaginationChange}
+        />
+      )}
+    </div>
+  );
+};
+
+Table.propTypes = {
+  columns: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
+  rows: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
+  emptyState: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+  pagination: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+  bodyMessage: PropTypes.node,
+  children: PropTypes.node,
+  itemCount: PropTypes.number,
+  onPaginationChange: PropTypes.func,
+};
+
+Table.defaultProps = {
+  emptyState: undefined,
```

Comment:
do we need undefined default value? it's undefined anyway.

Reply:
I believe yes.

1. It easier to understand the code when you can actually see the default values.
2. To satisfy eslint :)